src/ProgressOnderwijsUtils/Pocos/BulkInsertTarget.cs

```diff
@@ -40,10 +47,59 @@ public BulkInsertTarget With(BulkCopyFieldMappingMode mode)
         public BulkInsertTarget With(SqlBulkCopyOptions options)
             => new BulkInsertTarget(TableName, Columns, Mode, options);
 
+        [SuppressMessage("ReSharper", "PossibleMultipleEnumeration")]
+        static (T[] head, IEnumerable<T> fullSequence) PeekAtPrefix<T>(IEnumerable<T> enumerable, int firstN)
+        {
+            var enumerator = enumerable.GetEnumerator();
+            var buffer = new List<T>();
+            for (var i = 0; i < firstN; i++) {
+                if (enumerator.MoveNext()) {
+                    buffer.Add(enumerator.Current);
+                } else {
+                    var partialHead = buffer.ToArray();
+                    return (partialHead, partialHead);
+                }
+            }
+            var completeHead = buffer.ToArray();
+            if (enumerable is IReadOnlyList<T> || enumerable is ICollection<T>) {
+                return (completeHead, enumerable);
+            }
+
+            IEnumerable<T> FullSequence()
+            {
+                if (completeHead.PretendNullable() != null) {
+                    foreach (var item in completeHead) {
+                        yield return item;
+                    }
+#pragma warning disable IDE0059 // Unnecessary assignment of a value
+                    completeHead = null;
+#pragma warning restore IDE0059 // Unnecessary assignment of a value
+                } else {
+                    enumerator = enumerable.GetEnumerator();
+                }
+
+                while (enumerator.MoveNext()) {
+                    yield return enumerator.Current;
+                }
+                enumerator.Dispose();
+            }
+
+            return (completeHead, FullSequence());
+        }
+
         public void BulkInsert<[MeansImplicitUse(ImplicitUseKindFlags.Access, ImplicitUseTargetFlags.WithMembers)]
             T>(SqlConnection sqlConn, IEnumerable<T> pocos, CommandTimeout timeout = default, CancellationToken cancellationToken = default)
             where T : IReadImplicitly
         {
+            if (Options == defaultBulkCopyOptions) {

```

Comment:
Dit als explicitie parameter?

Reply:
Dit snap ik niet helemaal - wat wil je als expliciete parameter?  De defaults zijn echt de defaults; dat is een constante, en de Options, zijn whatever je hier hebt ingesteld (conceptueel een parameter, maar dit is een builder),

Reply:
Ik bedoel de hele bool als parameter. Ik vind de afhankelijkheid van een specifieke set Options niet heel voor de hand liggend

Reply:
Ik heb het iets anders opgelost, maar ik conceptueel wil ik dit checken dichtbij de small-batch implementatie, want die heeft kennis van welke opties die ondersteund.  Als ik hier ergens ooit een bool van zou maken, dan moet de aanroeper weten wat allemaal wel werkt, en wat niet, en dat lijkt me tricky.  Zo is het een implementatie-detail die alleen relevant is als optimalisatie.

Reply:
Maar nu is het geschreven als een afhankelijkeheid van de defaultBulkCopyOptions, terwijl die afhankelijkheid er niet is

Reply:
Ik zou `defaultBulkCopyOptions` `optionsThatMirrorSingleRowInsertIntoBehavior` oid; maar iig die afhankelijkheid is niet toevallig.  De defaults zijn gekozen om het verschil tussen `insert into` en bulk-copy zo klein mogelijk te houden, en deze optimalisatie werkt alleen als de code aan kan nemen dat insert into en bulk copy hetzelfde doen (bijv. in termen van fire triggers enzo).

Reply:
renamed.